### PR TITLE
gh-91565 Simple changes to reflect new bug reporting location. 

### PR DESCRIPTION
**Pull request by GadgetSteve**
# gh-91565 Simple changes to reflect new bug reporting location. 

Changes to code & document files to specify the GitHub link for issues rather than the BPO location.
Does not address any changes to Doc/tools/extensions/pyspecific.py #91565

### CODE REVIEW

File: Doc/README.rst [hunk 1]
```diff
@@ -130,7 +130,7 @@ Contributing
 ============
 
 Bugs in the content should be reported to the
-`Python bug tracker <https://bugs.python.org>`_.
+`Python bug tracker <https://github.com/python/cpython/issues>`_.
```
Comment (hugovk) [inline Doc/README.rst]: Let's be consistent with [devguide](https://devguide.python.org/tracker/) and call it the "issue tracker":

```suggestion
`Python issue tracker <https://github.com/python/cpython/issues>`_.
```

Comment (hugovk) [inline Doc/README.rst]: And in fact, shall we link to https://devguide.python.org/tracker/ instead? 

That has info about checking issues don't already exist, might help avoid duplicates.

```suggestion
`Python issue tracker <https://devguide.python.org/tracker/>`_.
```

File: Lib/locale.py [hunk 2]
```diff
@@ -1475,7 +1475,7 @@ def getpreferredencoding(do_setlocale=True):
 # to include every locale up to Windows Vista.
 #
 # NOTE: this mapping is incomplete.  If your language is missing, please
-# submit a bug report to the Python bug tracker at http://bugs.python.org/
+# submit a bug report to the Python bug tracker at https://github.com/python/cpython/issues
```
Comment (hugovk) [inline Lib/locale.py]:
```suggestion
# submit a bug report to the Python bug tracker: https://devguide.python.org/tracker/
```

File: Lib/platform.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 """
 #    This module is maintained by Marc-Andre Lemburg <mal@egenix.com>.
 #    If you find problems, please submit bug reports/patches via the
-#    Python bug tracker (http://bugs.python.org) and assign them to "lemburg".
+#    Python bug tracker (https://github.com/python/cpython/issues) and
```
Comment (hugovk) [inline Lib/platform.py]:
```suggestion
#    Python bug tracker (https://devguide.python.org/tracker/) and
```

File: Python/dtoa.c [hunk 4]
```diff
@@ -72,7 +72,7 @@
 /* Please send bug reports for the original dtoa.c code to David M. Gay (dmg
  * at acm dot org, with " at " changed at "@" and " dot " changed to ".").
  * Please report bugs for this modified version using the Python issue tracker
- * (http://bugs.python.org). */
+ * (https://github.com/python/cpython/issues). */
```
Comment (hugovk) [inline Python/dtoa.c]:
```suggestion
 * (https://devguide.python.org/tracker/). */
```

File: Mac/README.rst [hunk 5]
```diff
@@ -334,9 +334,9 @@ The configure script sometimes emits warnings like the one below::
    configure: WARNING: libintl.h:     section "Present But Cannot Be Compiled"
    configure: WARNING: libintl.h: proceeding with the preprocessor's result
    configure: WARNING: libintl.h: in the future, the compiler will take precedence
-   configure: WARNING:     ## --------------------------------------- ##
-   configure: WARNING:     ## Report this to https://bugs.python.org/ ##
-   configure: WARNING:     ## --------------------------------------- ##
+   configure: WARNING:     ## ------------------------------------------------------- ##
+   configure: WARNING:     ## Report this to https://github.com/python/cpython/issues ##
```
Comment (hugovk) [inline Mac/README.rst]:
```suggestion
   configure: WARNING:     ## Report this to https://devguide.python.org/tracker/ ##
```

**Review by hugovk — COMMENTED** [review]
Thanks, a few suggestions.

File: Doc/README.rst (hunk 1 above)
Comment (GadgetSteve) [reply Doc/README.rst]: Much better!

Comment (GadgetSteve) [reply Doc/README.rst]: Superseded by later comment

**Comment by arhadthedev** [comment]
`configure.ac` should be changed too; I created a PR in your fork to address this: https://github.com/GadgetSteve/cpython/pull/6.

**Comment by ezio-melotti** [comment]
While I agree that in some places pointing to the Devguide is better than pointing directly here, I think the wording surrounding those links should be updated accordingly.

For example instead of saying "Bugs in the content should be reported to the [Python issue tracker](https://devguide.python.org"/tracker/)", I would either say something like "See [the devguide](https://devguide.python.org"/tracker/) for information about reporting a bug." or just include both links.

**Comment by GadgetSteve** [comment]
> While I agree that in some places pointing to the Devguide is better than pointing directly here, I think the wording surrounding those links should be updated accordingly.
> 
> For example instead of saying "Bugs in the content should be reported to the [Python issue tracker](https://devguide.python.org%22/tracker/)", I would either say something like "See [the devguide](https://devguide.python.org%22/tracker/) for information about reporting a bug." or just include both links.

I quite agree with rewording and will amend correctly @ezio-melotti - thinking about it I don't think that including both links is ideal as:

 a) The devguide mentions checking if the issue has already been reported
 b) The devguide also gives guidance on what to include in the report & on the whole process
 c) Including **both** links will give someone twice the work in the event of any future move of the issue tracker

So unless there are compelling reasons to include both links I will just update to clarify that the devguide should be followed for reporting issues.

**Comment by hugovk** [comment]
That's better, thanks.

Need to check the CI failure:
```
Generated files not up to date.
Perhaps you forgot to run make regen-all or build.bat --regen. ;)
```

**Comment by GadgetSteve** [comment]
Note that rewording of the bug reporting references in configure are reverted by autoconfig - it may be possible to redefine this wording but it is not clear how.

**Comment by GadgetSteve** [comment]
> That's better, thanks.
> 
> Need to check the CI failure:
> 
> ```
> Generated files not up to date.
> Perhaps you forgot to run make regen-all or build.bat --regen. ;)
> ```

Those were the commands that I was missing to clear the error - unfortunately it reverts the rewording in configure and I can't see how to override it.

**Comment by hugovk** [comment]
@arhadthedev Do you know how to do it?

File: Lib/platform.py [hunk 6]
```diff
@@ -9,8 +9,9 @@
 
 """
 #    This module is maintained by Marc-Andre Lemburg <mal@egenix.com>.
-#    If you find problems, please submit bug reports/patches via the
-#    Python bug tracker (http://bugs.python.org) and assign them to "lemburg".
+#    If you find problems, please submit bug reports/patches as detailed
+#    in the Python devguide (https://devguide.python.org/tracker/) and
+#    assign them to "lemburg".
```
Comment (ronaldoussoren) [inline Lib/platform.py]: "lemburg" should be "malemburg" here.

**Review by ronaldoussoren — COMMENTED** [review]
I don't have anything to add other than a minor change to Lib/platform.py due to Marc-Andre's GH user name being different than his Roundup user name.

The change to Mac/README.rst looks good to me, as do the other changes.

File: Lib/platform.py (hunk 6 above)
Comment (GadgetSteve) [reply Lib/platform.py]: @ronaldoussoren Thanks for pointing that out correction being pushed in a few minutes.

**Comment by GadgetSteve** [comment]
Now gives:

```
#                                                                   
# Report bugs as detailed at <https://devguide.python.org/tracker/>.
#  
``` 
In `configure` header but still says:
```
 Report bugs to <https://devguide.python.org/tracker/>
```
Later.

**Comment by ezio-melotti** [comment]
The links should now be updated to point to https://devguide.python.org/triage/issue-tracker/

**Comment by encukou** [comment]
It seems we forgot about this PR, apologies for that. Most of the links have been fixed in the mean time.

I hope you don't mind me picking this up, and pushing directly.

File: Lib/platform.py (hunk 3 above)
Comment (ezio-melotti) [reply Lib/platform.py]: This was marked as solved, but the suggestion hasn't been committed.  Not sure if it was intentional or an oversight.

I see two options here.  If we want to keep a direct link to the issue tracker, we can just tweak the original sentence
```suggestion
#    Python issue tracker (https://github.com/python/cpython/issues) and
```

If we want to link to the devguide as @hugovk was suggesting, the comment should be rephrased a bit and the devguide URL updated.

Comment (encukou) [reply Lib/platform.py]: For the target audience here, I don't think the devguide adds much. The main info is who to tag.

**Pull request merged**